context.go

```diff
@@ -0,0 +1,69 @@
+package api2go
+
+import (
+	"golang.org/x/net/context"
+	"time"
+)
+
+// APIContext api2go context for handlers, nil implementations related to Deadline and Done.
+type APIContext struct {
```

Comment:
Ah, I got ya. That's an easy one then ;-)

That would imply we want to allow people to provide a New func to the pool then if they choose to implement the interface, so I'll add that at the same time.